odoo/addons/base/tests/test_cli.py

```diff
@@ -60,3 +69,30 @@ def test_help(self):
                     if line.startswith("   ") and (result := re.search(r'    (\w+)\s+(\w.*)$', line)):
                         actual.add(result.groups()[0])
                 self.assertGreaterEqual(actual, expected, msg="Help is not showing required commands")
+
+    @unittest.skipIf(os.name != 'posix', '`os.openpty` only available on POSIX systems')
+    def test_shell(self):
+
+        main, child = os.openpty()
+
+        with tempfile.NamedTemporaryFile(mode="w+", encoding="utf-8") as startup_file:
+            startup_file.write("shell_name = f'Hello from {shell_name}'")
+            startup_file.flush()
+
+            shell = self.popen_command(
+                'shell',
+                '--shell-interface=python',
+                f'--shell-file={startup_file.name}',
+                stdin=main,
+                close_fds=True,
+            )
+
+            with os.fdopen(child, 'w', encoding="utf-8") as stdin_file:
+                stdin_file.write(
+                    'print(shell_name)\n'
+                    'exit()\n'
+                )
+            shell.wait()
+
+        actual = shell.stdout.readlines()[-2].rstrip()
+        self.assertEqual(actual, ">>> Hello from python")
```

Comment:
```suggestion
        from odoo.tests import Like
        self.assertEqual(shell.stdout.read().splitlines(), [
            Like("env: <odoo.api.Environment object at 0x...>"),
            Like("odoo: <module 'odoo' from '/.../odoo/__init__.py'>"),
            Like("openerp: <module 'odoo' from '/.../odoo/__init__.py'>"),
            "self: res.users(1,)",
            "Python 3...",
            'Type "help", "copyright", "credits" or "license" for more information.',
            "(Console)",
            ">>> Hello from python",
        ])
```
I used `read().splitlines()` to discard the \n of every line. 
I used `Like` which is a tool from our test that simplifies string-comparison with assertions, it let you use `...` to mean "anything here".

btw I didn't test that code, maybe it doesn't work :grimacing:

Reply:
mkay 👍🏻 
I like the `Like` (no pun intended)